pipeline-model-api/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/ast/ModelASTAxes.java

```diff
@@ -0,0 +1,121 @@
+package org.jenkinsci.plugins.pipeline.modeldefinition.ast;
+
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+import org.jenkinsci.plugins.pipeline.modeldefinition.validator.ModelValidator;
+
+import javax.annotation.Nonnull;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.List;
+
+/**
+ * Represents a map of axis types to axis names (i.e., the name of the configured installation). Corresponds to
+ * {@code axes}.
+ *
+ * @author Andrew Bayer
+ */
+public final class ModelASTAxes extends ModelASTElement {
+    private Map<ModelASTKey, List<ModelASTValue>> axes = new LinkedHashMap<>();
+
+    public ModelASTAxes(Object sourceLocation) {
+        super(sourceLocation);
+    }
+
+    @Override
+    public JSONArray toJSON() {
+        final JSONArray a = new JSONArray();
+        for (Map.Entry<ModelASTKey, List<ModelASTValue>> entry: axes.entrySet()) {
+            JSONObject o = new JSONObject();
+            o.accumulate("key", entry.getKey().toJSON());
+
+            for(ModelASTValue value: entry.getValue()) {
+                o.accumulate("value", value.toJSON());
+            }
+
+            a.add(o);
+        }
+        return a;
+    }
+
+    @Override
+    public void validate(@Nonnull final ModelValidator validator) {
+        validator.validateElement(this);
+        for (Map.Entry<ModelASTKey, List<ModelASTValue>> entry : axes.entrySet()) {
+            entry.getKey().validate(validator);
+
+            for(ModelASTValue value: entry.getValue()) {
+                value.validate(validator);
+            }
+        }
+    }
+
+    @Override
+    public String toGroovy() {
+        StringBuilder result = new StringBuilder("axes {\n");
+        for (Map.Entry<ModelASTKey, List<ModelASTValue>> entry : axes.entrySet()) {
+            result.append(entry.getKey().toGroovy()).append(" (");
+            String comma = " ";
```

Comment:
This is a little mind-bending - `comma` equals `" "`? =) I get it, but there's probably a better way to do this - especially with this moved on top of #227 and therefore being Java 8. I just started playing with streams for the first time (I know, pathetically late, but hey) and that's pretty much perfect for this sort of thing - like, say:

```
result.append(entry.getValue().stream().map(ModelASTValue::toGroovy).collect(Collectors.joining(", ")));
```